app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -2,19 +2,77 @@
 
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
 import android.net.Uri;
 
 import org.schabi.newpipe.R;
 
 public final class ShareUtils {
-    private ShareUtils() { }
+    private ShareUtils() {
+    }
 
+    /**
+     * Open the url with the system default browser.
+     * <p>
+     * If no browser is set as default, fallbacks to
+     * {@link ShareUtils#openUrlBrowserChooser(Context, String)}
+     *
+     * @param context the context to use
+     * @param url     the url to browse
+     */
     public static void openUrlInBrowser(final Context context, final String url) {
-        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+        final Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(url));
+
+        final String defaultBrowserPackageName = getDefaultBrowserPackageName(context);
+
+        if (defaultBrowserPackageName.equals("android")) {
+            // no browser set as default
+            openUrlBrowserChooser(context, url);
+        } else {
+            intent.setPackage(defaultBrowserPackageName);
```

Comment:
Move the intent instantiation code inside the else branch, as it is not needed otherwise